### ISSUE TITLE
CSVReader crashes when reading an empty file

### ISSUE THREAD

**Issue by Krensi** [hardbyte/python-can#771]
The next statement in the __iter__ function of the class CSVReader is not inside the for loop. There for a Stopiteration error is not catched.

Rewriting the code like this fixes this:
```python
def __iter__(self):
        for (index, line) in enumerate(self.file):
            if index > 0:
                timestamp, arbitration_id, extended, remote, error, dlc, data = line.split(',')

                yield Message(
                    timestamp=float(timestamp),
                    is_remote_frame=(remote == '1'),
                    is_extended_id=(extended == '1'),
                    is_error_frame=(error == '1'),
                    arbitration_id=int(arbitration_id, base=16),
                    dlc=int(dlc),
                    data=b64decode(data),
                )

        self.stop()
```

**Comment by bessman** [comment]
This is [fixed](https://github.com/hardbyte/python-can/blob/b87329983d03b91b16920d47a8b4f90e650ca433/can/io/csv.py#L91) in develop:
```       
        # skip the header line
        try:
            next(self.file)
        except StopIteration:
            # don't crash on a file with only a header
            return
```

**Comment by karlding** [comment]
Closing since the fix is released as part of [v3.3.3](https://github.com/hardbyte/python-can/tree/3.3.3).
